src/Microsoft.AspNetCore.Hosting/Internal/StartupLoader.cs

```diff
@@ -143,9 +143,8 @@ IServiceProvider ConfigureServicesWithContainerConfiguration(IServiceCollection
                     {
                         // Get the default factory
                         var serviceProviderFactory = HostingServiceProvider.GetRequiredService<IServiceProviderFactory<IServiceCollection>>();
-
-                        // Don't bother calling CreateBuilder since it just returns the default service collection
-                        applicationServiceProvider = serviceProviderFactory.CreateServiceProvider(services);
+                        var builder = serviceProviderFactory.CreateBuilder(services);
```

Comment:
Is it guaranteed that factory exists?

Reply:
This has been the case since 2.0.  There’s always one registered. Unless somebody removes it.

Reply:
*shrug* it's a weird use of embedded methods.